src/librustdoc/html/static/js/search.js

```diff
@@ -2749,6 +2751,15 @@ class DocSearch {
                     return a - b;
                 }
 
+                // in type based search, put functions first
+                if (parsedQuery.hasReturnArrow) {
+                    a = (aaa.item.ty !== TY_FN && a.item.ty !== TY_METHOD);
+                    b = (bbb.item.ty !== TY_FN && b.item.ty !== TY_METHOD);
```

Comment:
Sorry. I got this wrong...

```suggestion
                    a = (aaa.item.ty !== TY_FN && aaa.item.ty !== TY_METHOD);
                    b = (bbb.item.ty !== TY_FN && bbb.item.ty !== TY_METHOD);
```